Hexastack/Hexabot
# fix: smtp config

# Motivation

In this PR, I did the following changes : 
- Removed SMTP config from settings (now only passed as env var)
- MailerService is an optional provider (available only if enabled in config)
- smtp4dev moved to a seperate docker compose file (will need to do `make start SMTP4DEV=1` to run it)
- Fixed email templates and i18n translation for user registration, confirmation, and password reset
- Minor other changes

Documentation has been updated here : https://docs.hexabot.ai/developer-guide/smtp-configuration-and-emails

Fixes #37 

## How to test

1. Test without SMTP (EMAIL_SMTP_ENABLED=false) : `make start` and then try inviting a user. The app will work normally as if it's able to send emails. In the background it will not.

2. Test with smtp4dev (EMAIL_SMTP_ENABLED=true) : `make start SMTP4DEV=1`   and then try inviting a user. Access http://localhost:9002/ and check if email is received. Try changing the default lang to "fr" and re-test to see if the email is received in french. Do the same for other scenarios : Confirmation email after signup and password reset email. 

2. Test with a 3rd party SMTP server (EMAIL_SMTP_ENABLED=true + use mailtrap.io or gmail) : Ensure the setup the env var before `make start` : 
```
EMAIL_SMTP_ENABLED=true
EMAIL_SMTP_HOST=sandbox.smtp.mailtrap.io
EMAIL_SMTP_PORT=2525
EMAIL_SMTP_SECURE=false
EMAIL_SMTP_USER=***************
EMAIL_SMTP_PASS=***************
EMAIL_SMTP_FROM=from@example.com
```
## Type of change

Please delete options that are not relevant.


# Checklist:

- [x] I have made corresponding changes to the documentation (Storybook)
